src/lib.rs

```diff
@@ -315,24 +479,33 @@ macro_rules! impl_trait(
                 }
             }
         }
+
+        unsafe impl FromByteVec for $to {
+            fn from_byte_vec(mut vec: Vec<u8>) -> Result<Vec<$to>, Error> {
+                let (len, capacity) = check_vec_constraints::<$to>(&vec)?;
+                let ptr = vec.as_mut_ptr();
+                mem::forget(vec);
+                unsafe {
+                    Ok(Vec::from_raw_parts(ptr as *mut $to, len, capacity))
+                }
+            }
+        }
+
+        impl IntoByteVec for Vec<$to> {
```

Comment:
Here `IntoByteVec` is implemented for `Vec<$to>` inside macro.

Reply:
:+1:

Reply:
So, except for additional test cases there isn't anything left right?

Reply:
And the docs comments I just did :) What additional tests were you thinking of?

But yes, the code itself is good like this. Thanks a lot!

Reply:
For errors, Alignment, Length, Capacity mismatch.

Reply:
:+1: